plutus-metatheory/src/Untyped/RenamingSubstitution.lagda.md

```diff
@@ -54,19 +61,30 @@ lift-cong : ∀{X Y}
 lift-cong p nothing  = refl
 lift-cong p (just x) = cong just (p x)
 
+renList-cong : ∀{X Y}
+    {ρ ρ' : Ren X Y}
+  → (∀ x → ρ x ≡ ρ' x)
+  → (t : List (X ⊢))
+  → renList ρ t ≡ renList ρ' t
+
 ren-cong : ∀{X Y}
     {ρ ρ' : Ren X Y}
   → (∀ x → ρ x ≡ ρ' x)
   → (t : X ⊢)
   → ren ρ t ≡ ren ρ' t
-ren-cong p (` x)       = cong ` (p x)
-ren-cong p (ƛ t)       = cong ƛ (ren-cong (lift-cong p) t)
-ren-cong p (t · u)     = cong₂ _·_ (ren-cong p t) (ren-cong p u)
-ren-cong p (force t)   = cong force (ren-cong p t)
-ren-cong p (delay t)   = cong delay (ren-cong p t)
-ren-cong p (con c)     = refl
-ren-cong p (builtin b) = refl
-ren-cong p error       = refl
+ren-cong p (` x)         = cong ` (p x)
+ren-cong p (ƛ t)         = cong ƛ (ren-cong (lift-cong p) t)
+ren-cong p (t · u)       = cong₂ _·_ (ren-cong p t) (ren-cong p u)
+ren-cong p (force t)     = cong force (ren-cong p t)
+ren-cong p (delay t)     = cong delay (ren-cong p t)
+ren-cong p (con c)       = refl
+ren-cong p (builtin b)   = refl
+ren-cong p error         = refl
+ren-cong p (constr i xs) = cong (constr i) (renList-cong p xs)
+ren-cong p (case t ts)   = cong₂ case (ren-cong p t) (renList-cong p ts)
+
+renList-cong p [] = refl
+renList-cong p (x ∷ xs) = cong₂ _∷_ (ren-cong p x) (renList-cong p xs) 
```

Comment:
Somehow seems like you should be able to get this automatically by the proof-equivalent of `fmap`, but whatever.